### DISCUSSION TITLE
How to configure multiple firebase projects in a single app ?

### DISCUSSION THREAD

**Discussion by kthilagarajan-agsft** [invertase/react-native-firebase#7220] [Q&A]
I am having 2 firebase projects, one for authentication and one for analytics.

I am having multiple apps which uses common authentication system. I don't want analytics to be part of the common project. So I created a separate firebase project for analytics. 

How do I configure? 

Since in the react native firebase project setup, we have to add google-service.json/plist in the project, right now I couldn't add both the projects. If I add like FirebaseAuth/google-services.plist and FirebaseAnalytics/google-services.plist, I am getting error. 

How to handle this?

Please share if any documentation available.

This documentation is from firebase, it provides help for native:
https://firebase.google.com/docs/projects/multiprojects#use_multiple_projects_in_your_application

**Comment by zulqarnain-empg** [comment]
any solution on this?

**Comment by zulqarnain-empg** [comment]
or any other way to accommodate multiple firebase in a single react native app.

**Reply by carlosen14** [reply]
One initialization (the default) must be done with those config files, and the other should be done on code at runtime, like this, all this information can be obtained from the same config files

```js
firebaseApp.initializeApp({
  appId,
  apiKey,
  projectId,
  databaseURL,
  messagingSenderId,
  storageBucket
}, {
  name: 'GA4'
})
```

at the end, you can specify a "name" so you can get the instance later

```js
firebase.app('GA4').messaging().foo()
```

if you don't specify a name, you will be using the default instance

```js
firebase.app().messaging().foo()
//or
messaging().foo()
```

**Reply by carlosen14** [reply]
but this doesn't work for analytics, as this only works with the default firebase app, otherwise will throw

**Reply by mikehardy** [reply]
Indeed, only some of the native modules support "secondary apps", and some do not.

analytics, perf, crashlytics - these all have significant things they do before secondary apps could be configured (that is, handle crashes and performance tracing and analytics events that are related to app startup...) so they cannot have secondary apps in a meaningful way

If that's a constraint of the project, you'll need to use firebase-js-sdk for those items.

**Reply by carlosen14** [reply]
Hey @mikehardy, it seems neither messaging nor dynamic links support multiple firebase apps, the app is throwing an exception with that message.. It says "Error: You attempted to call "firebase.app('CUSTOM_APP_NAME').messaging" but; messaging does not support multiple Firebase Apps.
Ensure you access messaging from the default application only."

is there anything we have missed? or was just a modification on the new SDK version?

**Reply by mikehardy** [reply]
I believe docs at firebase.io have a list of which modules support secondary apps and which don't. Messaging definitely does not. I don't think links does either, as the native SDKs do not for those modules
